fsspec/utils.py

```diff
@@ -266,11 +295,14 @@ def read_block(f, offset, length, delimiter=None, split_before=False):
         length = end - start
 
     f.seek(offset)
-    b = f.read(length)
+    if length is not None:
+        b = f.read(length)
+    else:
+        b = f.read()
```

Comment:
I assume we needed to do this (otherwise `if delimiter is None and length is None` I'm pretty sure this crashes), but happy to revert this.

Reply:
Let's not change the logic in this PR. I am not immediately sure about what you suggest.

Reply:
Sure -- can revert and add a `type: ignore` instead.